app/gateways/http/authentication/middleware_test.go

```diff
@@ -0,0 +1,168 @@
+		request.Header = badHeader
+		authenticationMiddleware := handler.Authenticate(nextHandlerFunc)
+		authenticationMiddleware(response, request)
+
+		got := response.Body.String()
+		expected := `{"error":"empty authorization header"}`
+
+		assert.Equal(t, http.StatusUnauthorized, response.Code)
+		assert.Equal(t, expected, strings.TrimSpace(got))
+
+	})
+
+	t.Run("should return 401 and error when access token is empty", func(t *testing.T) {
+		badHeader := http.Header{
+			"Authorization": []string{"Bearer"},
+		}
+		handler := createMiddlewareFakeHandler(nil)
+		nextHandlerFunc := createFakeHandlerFunc()
+		request, _ := http.NewRequest(http.MethodGet, "/transfers", nil)
+		response := httptest.NewRecorder()
+		request.Header = badHeader
+		authenticationMiddleware := handler.Authenticate(nextHandlerFunc)
+		authenticationMiddleware(response, request)
+
+		got := response.Body.String()
+		expected := `{"error":"empty token"}`
+
+		assert.Equal(t, http.StatusUnauthorized, response.Code)
+		assert.Equal(t, expected, strings.TrimSpace(got))
+	})
+
+	t.Run("should return 401 and error when the authentication method is wrong", func(t *testing.T) {
+		badHeader := http.Header{
+			"Authorization": []string{"auth "},
+		}
+		handler := createMiddlewareFakeHandler(nil)
+		nextHandlerFunc := createFakeHandlerFunc()
+		request, _ := http.NewRequest(http.MethodGet, "/transfers", nil)
+		response := httptest.NewRecorder()
+		request.Header = badHeader
+		authenticationMiddleware := handler.Authenticate(nextHandlerFunc)
+		authenticationMiddleware(response, request)
+
+		got := response.Body.String()
+		expected := `{"error":"invalid auth method"}`
+
+		assert.Equal(t, http.StatusUnauthorized, response.Code)
+		assert.Equal(t, expected, strings.TrimSpace(got))
+	})
+
+	t.Run("should return 401 and error when token is invalid", func(t *testing.T) {
+		header := http.Header{
+			"Authorization": []string{"Bearer "},
+		}
+		err := usecase.ErrInvalidToken
+		handler := createMiddlewareFakeHandler(err)
+		nextHandlerFunc := createFakeHandlerFunc()
+		request, _ := http.NewRequest(http.MethodGet, "/transfers", nil)
+		request.Header = header
+		response := httptest.NewRecorder()
+		authenticationMiddleware := handler.Authenticate(nextHandlerFunc)
+		authenticationMiddleware(response, request)
+
+		got := response.Body.String()
+		expected := `{"error":"invalid token"}`
+
+		assert.Equal(t, http.StatusUnauthorized, response.Code)
+		assert.Equal(t, expected, got)
+	})
+}
+
+func createFakeHandlerFunc() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		val := r.Context().Value(http_server.ContextID)
+		if val == nil {
+			w.WriteHeader(http.StatusUnauthorized)
+			_, _ = w.Write([]byte(idNotPresent))
+			return
+		}
+		valStr, ok := val.(string)
+		if !ok {
+			w.WriteHeader(http.StatusUnauthorized)
+			_, _ = w.Write([]byte(notString))
+			return
+		}
+		if valStr != "642e0d44-9792-4d6f-9a04-b40186dddbef" {
```

Comment:
since you have interest in this value during the test cases, it should either be passed as a parameter for this function and the others interested or extracted in a constant

Reply:
Yes! I put it in a constant